store/db.go

```diff
@@ -246,3 +246,53 @@ func (o *DBCommentOperator) posted(ctx context.Context, e *lookout.ReviewEvent,
 
 	return count > 0, nil
 }
+
+// DBOrganizationOperator operates on an organization database store
+type DBOrganizationOperator struct {
+	organizationStore *models.OrganizationStore
+}
+
+// NewDBOrganizationOperator creates new DBOrganizationOperator using kallax as storage
+func NewDBOrganizationOperator(store *models.OrganizationStore) *DBOrganizationOperator {
+	return &DBOrganizationOperator{store}
+}
+
+var _ OrganizationOperator = &DBOrganizationOperator{}
+
+func (o *DBOrganizationOperator) getOrganization(ctx context.Context, provider string, orgID string) (*models.Organization, error) {
+	q := models.NewOrganizationQuery().FindByProvider(provider).FindByInternalID(orgID)
+	return o.organizationStore.FindOne(q)
+}
+
+// Save persists the given config, updating the current one if it exists
+// for the given (provider, orgID)
+func (o *DBOrganizationOperator) Save(ctx context.Context, provider string, orgID string, config string) error {
+	m, err := o.getOrganization(ctx, provider, orgID)
+	if err != nil && err != kallax.ErrNotFound {
+		return err
+	}
+
+	if err == kallax.ErrNotFound {
+		m = models.NewOrganization(provider, orgID, config)
+	} else {
+		m.Config = config
+	}
+
+	_, err = o.organizationStore.Save(m)
+	return err
+}
+
+// Config returns the stored config for the given (provider, orgID). If there
+// are no records in the DB, it returns "" without error.
+func (o *DBOrganizationOperator) Config(ctx context.Context, provider string, orgID string) (string, error) {
+	m, err := o.getOrganization(ctx, provider, orgID)
+	if err != nil && err != kallax.ErrNotFound {
+		return "", err
+	}
+
+	if err == kallax.ErrNotFound {
+		return "", nil
```

Comment:
Why do we want to return an empty config if there is no such Organization? Why is that better than returning a not found, and let the caller to decide?
How can the caller know if the stored config is empty, or there is no Org?

Reply:
The callers will be:
- the web backend to report the current default value
- the `Server.HandleReview`, to merge the default with the local config

My reasoning here is that I wanted a "helper" to get the the default config for a given organization, rather than a method to retrieve the raw content from the DB. They don't care if the organization has an entry in the DB, if the entry has an empty string stored, or if there is no entry; they just want the string content.

With that in mind, this method works like a `map`,. If the organization does not have a config set, you get the zero value, `""`. I wasn't sure if this "helper" behaviour should go in `db.go`, or if it's too "high level" and should go as a wrapper somewhere else. WDYT?

Reply:
I see that logic out of the DB package, but creating that other helper somewhere only to wrapp the functionality `if err == NotFound return ""`... Let's keep it here if you prefer so.

Reply:
@dpordomingo keep in mind this package / struct isn't direct mapping to DB. We have ORM (kallax for that). It implements [repository pattern](https://docs.microsoft.com/en-us/dotnet/standard/microservices-architecture/microservice-ddd-cqrs-patterns/infrastructure-persistence-layer-design#the-repository-pattern). I used name "operator" instead of "repository" to avoid confusion with git repositories (as it's more important concept for lookout)

Reply:
/aside;
I think that since it's a repository, it should focus only on accessing the data, instead of returning custom data to fit some client logic. 
>_Repositories are classes or components that encapsulate the logic required to access data sources._

But as I said, I'm ok if we keep it as it is, and avoid creating more wrappers somewhere.